### PR TITLE
No stacktrace check for build-android-app

### PR DESCRIPTION
**Pull request by blindpirate**
It has been flaky for a long time. Previously we tried to fix
it by using StackTraceSanitizingOutputNormalizer but that didn't work
because the stacktrace check failure happens before normalizing the
output. Now let's simply disable the check for this case.

### CODE REVIEW

File: subprojects/docs/src/docsTest/java/org/gradle/docs/samples/IntegrationTestSamplesExecutor.java [hunk 1]
```diff
@@ -71,6 +73,11 @@ protected int run(String executable, List<String> args, List<String> flags, Outp
             .withToolchainDetectionEnabled()
             .withArguments(filteredFlags)
             .withTasks(args);
+
+        if (flags.stream().anyMatch(f -> f.equals(NO_STACKTRACE_CHECK))) {
```
Comment (wolfs) [inline subprojects/docs/src/docsTest/java/org/gradle/docs/samples/IntegrationTestSamplesExecutor.java:77]: I think this should go up where we already extract the `warningMode` and the system property shouldn't be propagated to the actual build.

File: subprojects/docs/src/samples/build-organization/structuring-software-projects/tests/build-android-app.sample.conf [hunk 2]
```diff
@@ -1,6 +1,8 @@
 commands: [{
     executable: gradle
-    args: build -x lint -x lintVitalRelease
+    # There's a flaky stacktrace: java.io.FileNotFoundException: https://dl.google.com/android/repository/sys-img/android-desktop/sys-img2-1.xml
+    #  at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1896) ...
```
Comment (wolfs) [inline subprojects/docs/src/samples/build-organization/structuring-software-projects/tests/build-android-app.sample.conf:4]: I wonder if we should exclude this stacktrace in our built-in stacktrace checks like we do for other things. Though I suppose for now let's only exclude it for this sample.

File: subprojects/docs/src/docsTest/java/org/gradle/docs/samples/IntegrationTestSamplesExecutor.java (hunk 1 above)
Comment (blindpirate) [reply subprojects/docs/src/docsTest/java/org/gradle/docs/samples/IntegrationTestSamplesExecutor.java]: Actually I prefer doing a bit refactoring work to get rid of the loop.

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Pull request merged**